pom.xml

```diff
@@ -4,32 +4,206 @@
+                                </requireJavaVersion>
+                                <enforceBytecodeVersion>
+                                    <maxJdkVersion>${maven.enforcer.maxJdkVersion}</maxJdkVersion>
+                                    <ignoredScopes>
+                                        <ignoredScope>test</ignoredScope>
+                                    </ignoredScopes>
+                                    <excludes>
+                                        <exclude>org.jenkins-ci.main:jenkins-core</exclude>
+                                        <exclude>org.jenkins-ci.main:cli</exclude>
+                                        <exclude>org.jenkins-ci.main:jenkins-test-harness</exclude>
+                                        <exclude>org.jenkins-ci.main:remoting</exclude>
+                                        <exclude>org.kohsuke.stapler:stapler</exclude>
+                                        <exclude>org.kohsuke.stapler:stapler-groovy</exclude>
+                                        <exclude>org.kohsuke.stapler:stapler-jelly</exclude>
+                                        <exclude>org.kohsuke.stapler:stapler-jrebel</exclude>
+                                        <exclude>org.jenkins-ci:task-reactor</exclude>
+                                        <exclude>com.google.code.findbugs:annotations</exclude>
+                                    </excludes>
+                                </enforceBytecodeVersion>
+                                <bannedDependencies>
+                                    <excludes>
+                                        <exclude>javax.servlet:servlet-api</exclude>
+                                        <exclude>org.sonatype.sisu:sisu-guice</exclude>
+                                        <exclude>log4j:log4j:*:jar:compile</exclude>
+                                        <exclude>log4j:log4j:*:jar:runtime</exclude>
+                                        <exclude>commons-logging:commons-logging:*:jar:compile</exclude>
+                                        <exclude>commons-logging:commons-logging:*:jar:runtime</exclude>
+                                    </excludes>
+                                </bannedDependencies>
+                                <requireUpperBoundDeps>
+                                    <excludes>
+                                        <exclude>com.google.guava:guava</exclude>
+                                        <exclude>commons-logging:commons-logging</exclude>
+                                        <exclude>com.google.code.findbugs:jsr305</exclude>
+                                        <exclude>net.java.dev.jna:jna</exclude>
+                                    </excludes>
+                                </requireUpperBoundDeps>
+                            </rules>
+                        </configuration>
+                    </execution>
+                </executions>
+                <dependencies>
```

Comment:
Not sure if you need to include all this either. Just simplify the plugin configuration down to excluding log4j as a banned dependency. The configurations are merged dynamically.

Reply:
alright..